consensus/src/experimental/tests/commit_phase_tests.rs

```diff
@@ -0,0 +1,642 @@
+        ));
+
+        let commit_vote_with_bad_signature = CommitVote::new_with_signature(
+            signer.author(),
+            good_ledger_info,
+            signer.sign(random_commit_vote.ledger_info()),
+        );
+
+        assert!(matches!(
+            commit_phase
+                .process_commit_vote(&commit_vote_with_bad_signature)
+                .await,
+            Err(Error::VerificationError)
+        ));
+    });
+}
+
+fn test_commit_phase_process_commit_decision() {
+    let mut runtime = consensus_runtime();
+    let (
+        _commit_tx,
+        _msg_tx,
+        _commit_result_rx,
+        _self_loop_rx,
+        _safety_rules_container,
+        signers,
+        state_computer,
+        _validator,
+        mut commit_phase,
+    ) = prepare_commit_phase();
+
+    timed_block_on(&mut runtime, async move {
+        let signer = &signers[0];
+
+        let (vecblocks, li_sig) = prepare_executed_blocks(&signers[0], state_computer);
+
+        let good_ledger_info = LedgerInfo::new(
+            vecblocks.last().unwrap().block_info(),
+            li_sig.ledger_info().consensus_data_hash(),
+        );
+
+        commit_phase.set_blocks(Some(PendingBlocks::new(vecblocks, li_sig)));
+
+        let random_commit_decision = generate_random_commit_decision(signer);
+
+        assert!(matches!(
+            commit_phase
+                .process_commit_decision(&random_commit_decision)
+                .await,
+            Err(Error::InconsistentBlockInfo(_, _))
+        ));
+
+        let commit_decision_with_bad_signature =
+            CommitDecision::new(LedgerInfoWithSignatures::new(
+                good_ledger_info,
+                BTreeMap::<AccountAddress, Ed25519Signature>::new(),
+            ));
+
+        assert!(matches!(
+            commit_phase
+                .process_commit_decision(&commit_decision_with_bad_signature)
+                .await,
+            Err(Error::VerificationError)
+        ));
+    });
+}
+
+fn test_commit_phase_check_commit() {
+    let mut runtime = consensus_runtime();
+    let (
+        _commit_tx,
+        _msg_tx,
+        _commit_result_rx,
+        _self_loop_rx,
+        _safety_rules_container,
+        signers,
+        state_computer,
+        _validator,
+        mut commit_phase,
+    ) = prepare_commit_phase();
+
+    timed_block_on(&mut runtime, async move {
+        let _signer = &signers[0];
+
+        let (vecblocks, li_sig) = prepare_executed_blocks(&signers[0], state_computer);
+
+        assert!(commit_phase.blocks().is_none());
+
+        // when blocks is none
+        commit_phase.check_commit().await.ok();
+
+        assert!(commit_phase.blocks().is_none());
+
+        commit_phase.set_blocks(Some(PendingBlocks::new(vecblocks.clone(), li_sig.clone())));
+
+        // when blocks is good
+        commit_phase.check_commit().await.ok();
+
+        // the block should be consumed
+        assert!(commit_phase.blocks().is_none());
```

Comment:
why? this is ordered_ledger info not executed?

Reply:
resolved